download.php

```diff
@@ -36,18 +36,11 @@
 
 // Get all questions form database with their answers and last_used tag
 global $DB;
-$query = '
-    SELECT question.id "id", question.questiontext "question_text", GROUP_CONCAT(answers.answer) "answers", topictagged.lastused "last_used"
-    FROM {question} question
-        JOIN {question_answers} answers
-            ON answers.question = question.id
-        JOIN {question_topictagged} topictagged
-            ON question.id = topictagged.questionid
-    WHERE question.category = ' . $categoryid . '
-    GROUP BY question.id;
-';
+global $sql_question_answer_lastused;
+require('consts.php');
+$query = $sql_question_answer_lastused;
 
-$entries = $DB->get_records_sql($query);
+$entries = $DB->get_records_sql($query, ['categoryid' => strval($categoryid)]);
```

Comment:
This makes using the constant strings more complicated than required, and obfuscates the code a bit. You could create a function that can handle the query for you. That way, the `require` condition can be moved to top level and the function can be used anywhere. However, it makes more sense for strings / queries that are reused; since this query appears to be unique, it is not required for it to be moved.